File: test-examples/test-code.js
// Test JavaScript code for code analysis
import fs from 'fs'; // Unused import
import path from 'path';

function main() {
    // TODO: Add proper error handling
    console.log("This should use a proper logger");

    // Very long line that exceeds common recommendations and should be refactored
    const veryLongVariableName = "This is a very long string that demonstrates a line length issue in JavaScript code";

    // Multiple console.log statements
    console.log("Debug 1");
    console.log("Debug 2");
    console.log("Debug 3");

    // FIXME: This function is too complex
    if (true) {
        if (true) {
            if (true) {
                console.log("Deeply nested code");
            }
        }
    }

    // Trailing whitespace   
    const result = "value";   

    return result;
}

main();

File: test-examples/test-code.py
# Test Python code for code analysis
import os
import sys
import unused_module

def main():
    # TODO: Add proper error handling
    print("This should use logging instead")
    
    # Very long line that exceeds PEP 8 recommendations and should be refactored into smaller parts
    very_long_variable_name = "This is a very long string that demonstrates a line length issue in Python code"
    
    # Multiple print statements
    print("Debug 1")
    print("Debug 2")
    print("Debug 3")
    
    # FIXME: This function is too complex
    if True:
        if True:
            if True:
                print("Deeply nested code")
    
    # Trailing whitespace    
    result = "value"    
    
    return result

if __name__ == "__main__":
    main()
